### PR TITLE
Initial action extension samples

### PR DESCRIPTION
**Pull request by bbowyersmyth**
SimpleAction - Basic shell
InputOutputAction - Registering inputs and outputs
DocumentAction - How to handle generated documents

Note: These wont build until we have some nuget packages up.

### CODE REVIEW

**Review by Intelledox-Brad — APPROVED** [review]

File: ExtensionExamples/ActionExtension/DocumentAction.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            {
+                Result = Intelledox.QAWizard.Design.ActionResultType.Success
+            };
+
+            try
+            {
+                foreach (var doc in properties.Documents)
+                {
+                    using (var docStream = await properties.GetDocumentStreamAsync(doc))
+                    using (var destStream = new MemoryStream())
```
Comment (simopassfield) [inline ExtensionExamples/ActionExtension/DocumentAction.cs]: Inconsistent '{' '}' ?

File: ExtensionExamples/ActionExtension/DocumentAction.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            ActionResult results = new ActionResult()
+            {
+                Result = Intelledox.QAWizard.Design.ActionResultType.Success
+            };
+
+            try
+            {
+                foreach (var doc in properties.Documents)
+                {
+                    using (var docStream = await properties.GetDocumentStreamAsync(doc))
```
Comment (simopassfield) [inline ExtensionExamples/ActionExtension/DocumentAction.cs:33]: Inconsistent '{ }'?? Petty I know but the one below has it.

File: ExtensionExamples/ActionExtension/DocumentAction.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
+﻿using Intelledox.Extension.Action;
+using Intelledox.QAWizard;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace SampleActionExtensions
+{
+    /// <summary>
+    /// This sample demonstrates dealing within generated documents that have been sent into the action.
```
Comment (simopassfield) [inline ExtensionExamples/ActionExtension/DocumentAction.cs]: This sample demonstrates accessing and working with generated document binaries within an Action. The project’s administrator configures which document(s) should generated at the end of each workflow step and the final submission of the form.

File: ExtensionExamples/ActionExtension/SimpleAction.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
+﻿using Intelledox.Extension.Action;
```
Comment (simopassfield) [inline ExtensionExamples/ActionExtension/SimpleAction.cs:1]: HelloWorld Action?

**Review by simopassfield — COMMENTED** [review]
Very petty things!

File: ExtensionExamples/ActionExtension/DocumentAction.cs (hunk 3 above)
Comment (bbowyersmyth) [reply ExtensionExamples/ActionExtension/DocumentAction.cs]: Re-worded the description. Left the second part. Not sure if we should be including Design level setup in samples though. Doesn't hurt in this case.

File: ExtensionExamples/ActionExtension/DocumentAction.cs (hunk 2 above)
Comment (bbowyersmyth) [reply ExtensionExamples/ActionExtension/DocumentAction.cs:33]: It is common to keep usings at the same level. We don't have a style rule on it though.

**Review by simopassfield — APPROVED** [review]
Yeah I know design comments seem too much. But I often see developers writing actions who have no Infiniti experience asking questions like 'why were 0 documents passed to the action'

**Pull request merged**